pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -489,22 +500,22 @@ func (f *fqdnController) onDNSResponse(
 		}
 		// The FQDN will be added to the queue only after `lowestTTL` value which
 		// would already have been derived using the minTTL logic.
-		f.dnsQueryQueue.AddAfter(fqdn, time.Duration(lowestTTL)*time.Second)
+		f.dnsQueryQueue.AddAfter(fqdn, maxTimeToReQuery.Sub(currentTime))
```

Comment:
That doesn't seem right: the time after which we should automatically re-query should be a "min", not a "max"

Reply:
right , i mean we need Min because when any of these IP expires its is then that we have to send DNS queries . Got your point.